src/dialogs/quick-bar/ha-quick-bar.ts

```diff
@@ -181,16 +188,37 @@ export class QuickBar extends LitElement {
     }
 
     this._items = (this._commandMode ? this._commandItems : this._entityItems)
-      .filter(({ text, altText }) => {
-        const values = [text];
-        if (altText) {
-          values.push(altText);
-        }
-        return fuzzySequentialMatch(this._itemFilter, values);
-      })
-      .sort((itemA, itemB) => compare(itemA.text, itemB.text));
+      .filter(({ text, altText }) =>
+        altText
+          ? isPatternInWords(this._itemFilter, text, altText)
+          : isPatternInWords(this._itemFilter, text)
+      )
+      .map<ScoredItem>((item) => ({
+        item,
+        score: item.altText
+          ? fuzzySequentialMatch(this._itemFilter, item.text, item.altText)
+          : fuzzySequentialMatch(this._itemFilter, item.text),
+      }))
+      .sort(this.compareFuzzyMatches)
+      .map((scoredItem) => scoredItem.item);
   }
 
+  private compareFuzzyMatches = (a: ScoredItem, b: ScoredItem) => {
+    const scoreA = a.score;
+    const scoreB = b.score;
+
+    if (!scoreA || !scoreB) {
+      return 0;
+    }
+
+    const isNumber = !isNaN(Number(scoreA));
+    if (isNumber) {
+      return scoreA > scoreB ? -1 : 1;
```

Comment:
What if they are equal?

Reply:
N/A after recent push removing this compare entirely.

Reply:
Turns out I needed the sorting logic after all. Put back the compare